ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModel.kt

```diff
@@ -114,7 +114,10 @@ internal class PaywallViewModelImpl(
         val activity = context.getActivity() ?: error("Activity not found")
         when (val currentState = _state.value) {
             is PaywallViewState.Loaded -> {
-                purchasePackage(activity, currentState.selectedPackage.rcPackage)
+                val selectedPackage = currentState.selectedPackage
+                if (!selectedPackage.currentlySubscribed) {
```

Comment:
I think we need to pass in the currently subscribed product ids when creating the `TemplateConfiguration`, so this won't work yet... But I don't think it hurts either.

Reply:
Also, until we change the UI, should we log a message when the user is subscribed?